### ISSUE TITLE
Have Teleport the ability to configure additional app stanzas. 

### ISSUE THREAD

**Issue by moisesmejia1200** [gravitational/teleport#21363]
What would you like Teleport to do?
- We would like to use teleport configure to define additional app stanzas in a newly created or existing teleport.yaml file

What problem does this solve?
- We write infrastructure-as-code and our teleport node servers are built using cloud-init and other automated means. 
- We generate the required configurations, but formats change. Some of our services require us to deploy teleport.yaml files that include multiple apps. We generate some of these using teleport configure, but are forced right now to manually create additional app configurations, an error-prone and high-maintenance activity. A better approach is to use the native tooling and teleport cli to generate the configurations in a consistent, maintainable format, and to better support changes to teleport configuration file structure over time

If a workaround exists, please include it.
- we currently use lines like "echo '(correct \s offset)keyword:' >> /etc/teleport/east/apps.yaml and sed, awk and cat <<FILE>> to hack in our missing stanzas to existing files. We could centrally store the while files that we need created, but given that the first part of each file, node, or app is generated with teleport config so that the tokens and enrollment are handled, we'd still be adding a file segment to an existing file.

**Comment by strideynet** [comment]
Out of interest, is there any reason why https://goteleport.com/docs/application-access/guides/dynamic-registration/ is not a good fit for this scenario ? In my opinion, we are more likely to maintain compatibility of these resource schemas and this has the added advantage that they can pretty much deploy one standardised Teleport configuration across their fleet w/o the need to modify the config on boot.

**Comment by zmb3** [comment]
+1, I believe dynamic registration would work well with the Terraform provider too.

**Comment by arcaven** [comment]
> Out of interest, is there any reason why https://goteleport.com/docs/application-access/guides/dynamic-registration/ is not a good fit for this scenario ? In my opinion, we are more likely to maintain compatibility of these resource schemas and this has the added advantage that they can pretty much deploy one standardised Teleport configuration across their fleet w/o the need to modify the config on boot.

Yes. While the approach suggested is ideal, the use case here is systems and containers that are 1) built though automation and are subject to being rebuilt from original source based on any change to the standards, images or software, including a teleport release and 2) run outside of the organizational boundary, behind firewalls, across NAT boundaries. Specifically, ci/cd pipeline runners will never have tcp reachability to these nodes even if they had a service mesh and knew how to find them directly. These are nodes that connect into proxies from outside networks, and make their apps (if configured) available to users of the centralized clusters. 

For apps on nodes "within" the teleport cluster/local to the environment, using the terraform provider is ideal; however, this is for use cases where that is not possible. It would be ideal if we could use dynamic registration to detail an app configuration to be run on a specific remote node when it checks in, but I'm not aware of this being an option? Nothing in the illustrated examples allow the user to bind a central app config to be run on a remote node, such that connectivity for that app is always sourced from the indicated node.

**Comment by zmb3** [comment]
> Nothing in the illustrated examples allow the user to bind a central app config to be run on a remote node, such that connectivity for that app is always sourced from the indicated node.

You can accomplish this via labels (though the other issue about making it hard to generate a teleport.yaml with specific app labels would probably need to be fixed for this to be a viable workflow for you.

For illustration's sake though, you'd have terraform create a dynamic app registration for your app that sits behind some firewall, with a label like `location: behind-firewall-76`.

Then you'd deploy an app service behind that same firewall, and configure it to monitor for apps with that label:

```
app_service:
  enabled: yes
  resources:
  - labels: 
      location: behind-firewall-76
```

Again, I recognize this takes us full circle because the original request is about programmatically generating app service config, and the suggestion to use the same app service config across your deployments won't work here. Just wanted to point out that this is possible. Full docs at: https://goteleport.com/docs/application-access/guides/dynamic-registration/

In summary, if we solve #21364, then I believe the approach outlined above should work for you and you wouldn't need the ability to configure additional app stanzas. What do you think?

**Comment by arcaven** [comment]
> In summary, if we solve https://github.com/gravitational/teleport/issues/21364, then I believe the approach outlined above should work for you and you wouldn't need the ability to configure additional app stanzas. What do you think?

That would be a welcome and very powerful method, I think, and if I can replicate this, it will change how I do things. If I understand correctly, then, the app label would be matched with node label(s) in a dynamic app registration applied in the cluster, and the app would be served from any nodes connecting into the cluster that have the label(s) - AND - and we could manage creating apps with terraform provider against the central teleport cluster, rather than managing any apps in yaml files in the nodes themselves. I think you'd make me a very happy terraformer this way.

**Comment by strideynet** [comment]
In which case, I'll close this issue down and we can focus this discussion on #21364 :)
